src/objects.c

```diff
@@ -212,13 +212,20 @@ static CK_RV obj_add_to_pool(P11PROV_OBJ *obj)
     (void)MUTEX_UNLOCK(pool);
     /* ------------- LOCKED SECTION */
 
+    if (ret == CKR_OK) {
+        P11PROV_debug("Object added to pool (idx=%d, obj=%p)", obj->poolid,
+                      obj);
+    }
+
     return ret;
 }
 
 static void obj_rm_from_pool(P11PROV_OBJ *obj)
 {
     P11PROV_OBJ_POOL *pool;
     CK_RV ret;
+    const char *errstr = NULL;
+    int idx;
```

Comment:
It id not a problem to de-reference obj->poolid, so there is no need to reintroduce idx here.

Up can keep the debug message before the locked section given you removed all pool-> de-referencing.

Reply:
Yeah of course. Sorry it's quite late in Europe... :)

Reply:
Fixed

Reply:
Please never feel pressured to react immediately, I am ok waiting a bit to get things right.